src/backend/gporca/libgpopt/src/xforms/CXformCTEAnchor2TrivialSelect.cpp

```diff
@@ -99,9 +99,17 @@ CXformCTEAnchor2TrivialSelect::Transform(CXformContext *pxfctxt,
 	// child of CTE anchor
 	CExpression *pexprChild = (*pexpr)[0];
 	pexprChild->AddRef();
+	BOOL fake = true;
+
+	if (COperator::EopLogicalCTEAnchor != pexpr->Pop()->Eopid() ||
```

Comment:
removed

Reply:
Seems my second question was unclear.
Why does the check for `COperator::EopLogicalLeftOuterCorrelatedApply` is needed?

Reply:
> Why does the check for `COperator::EopLogicalLeftOuterCorrelatedApply` is needed?

Yes, it is needed, because CXformCTEAnchor2TrivialSelect is used not only for CTEs.